File: CegCRMAPI/CegCRMClient/src/types/product.ts
export interface Product {
  id: string;
  name: string;
  description: string;
  price: number;
  stockQuantity: number;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
  saleProducts?: SaleProduct[];
}

export interface CreateProduct {
  name: string;
  description: string;
  price: number;
  stockQuantity: number;
  isActive: boolean;
}

export interface UpdateProduct {
  id: string;
  data: Partial<CreateProduct>;
}

export interface SaleProduct {
  id: string;
  productId: string;
  saleId: string;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
} 

File: CegCRMAPI/CegCRMAPI/Core/CegCRMAPI.Domain/Entities/Product.cs
using CegCRMAPI.Domain.Entities.Common;
using System;
using System.Collections.Generic;

namespace CegCRMAPI.Domain.Entities
{
    public class Product : BaseEntity
    {
        public string Name { get; set; } = string.Empty;
        public string Description { get; set; } = string.Empty;
        public decimal Price { get; set; }
        public int StockQuantity { get; set; }
        public bool IsActive { get; set; } = true;

        // Navigation properties
        public ICollection<SaleProduct> SaleProducts { get; set; } = new List<SaleProduct>();
    }
} 